.gitignore

```diff
@@ -3,6 +3,11 @@ dox/html
 manual/build
 sandbox
 !sandbox/CMakeLists.txt
+CMakeFiles
+CTestTestfile.cmake
+cmake_install.cmake
+Makefile
+.vscode
```

Comment:
That looks like a fixup of in-source builds. Can you please remove this but keep the `.vscode` and make it a separate commit.

Reply:
well true, but `.vscode` is a common hidden directory for visual studio code which gains popularity. So I wouldn't mind but I have no strong feelings either.